packages/schematics/angular/application/index_spec.ts

```diff
@@ -16,108 +17,92 @@ describe('Application Schematic', () => {
     style: 'css',
     skipTests: false,
-    minimal: false,
   };
 
+  let workspaceTree: UnitTestTree;
+  beforeEach(() => {
+    workspaceTree = schematicRunner.runSchematic('workspace', workspaceOptions);
+  });
+
   it('should create all files of an application', () => {
     const options = { ...defaultOptions };
 
-    const tree = schematicRunner.runSchematic('application', options);
+    const tree = schematicRunner.runSchematic('application', options, workspaceTree);
     const files = tree.files;
-    expect(files.indexOf('/foo/.editorconfig')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/.angular-cli.json')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/.gitignore')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/karma.conf.js')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/package.json')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/protractor.conf.js')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/README.md')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/tsconfig.json')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/tslint.json')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/e2e/app.e2e-spec.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/e2e/app.po.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/favicon.ico')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/index.html')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/main.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/polyfills.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/styles.css')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/test.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/tsconfig.app.json')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/tsconfig.spec.json')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/typings.d.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/assets/.gitkeep')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/environments/environment.prod.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/environments/environment.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/app/app.module.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/app/app.component.css')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/app/app.component.html')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/app/app.component.spec.ts')).toBeGreaterThanOrEqual(0);
-    expect(files.indexOf('/foo/src/app/app.component.ts')).toBeGreaterThanOrEqual(0);
+    expect(files.indexOf('/projects/foo/environments/environment.prod.ts'))
+      .toBeGreaterThanOrEqual(0);
+    expect(files.indexOf('/projects/foo/environments/environment.ts')).toBeGreaterThanOrEqual(0);
```

Comment:
Most of these files are source files and should be inside `src/`, with the `app` folder still existing. Only configuration files (`karma.conf.js`, `tsconfig.app.json` and `tsconfig.spec.json`) remain outside source.

You can see the target structure here (replace `projects/app` with `projects/foo`): https://github.com/filipesilva/workspace-project/tree/master/projects/app